public/components/MDS/DataSourceMenuWrapper.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useState } from 'react';
+import { Route, Switch } from 'react-router-dom';
+import {
+  DataSourceManagementPluginSetup,
+  DataSourceOption,
+  DataSourceSelectableConfig,
+  DataSourceViewConfig,
+} from '../../../../../src/plugins/data_source_management/public';
+import { AppMountParameters, CoreStart } from 'opensearch-dashboards/public';
+import { ROUTES } from '../../utils/constants';
+
+export interface DataSourceMenuWrapperProps {
+  core: CoreStart;
+  dataSourceManagement?: DataSourceManagementPluginSetup;
+  setHeaderActionMenu: AppMountParameters['setHeaderActionMenu'];
+}
+
+export const DataSourceMenuWrapper: React.FC<DataSourceMenuWrapperProps> = ({
+  core,
+  dataSourceManagement,
+  setHeaderActionMenu,
+}) => {
+  if (!dataSourceManagement) {
+    return null;
+  }
+
+  const [activeOption, setActiveOption] = useState({ id: '', label: '', checked: '' });
```

Comment:
what is `checked` state for?

Reply:
we can have another state `loading` to ensure that dataSources are fully loaded before rendering a component.

Reply:
That is present in the [`DataSourceOption`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source_management/public/components/data_source_menu/types.ts#L15) interface to added it here to be consistent.